### ISSUE TITLE
Add dependecies for PHP extensions

### ISSUE THREAD

**Issue by aschempp** [contao/core-bundle#533]
Should we add composer dependencies for the required PHP extensions?
https://docs.contao.org/books/manual/4.2/en/01-installation/configuring-the-live-server.html

Also `require-dev` for the optional ones?

**Comment by leofeyer** [comment]
I actually had that in one of the early 4.0 beta versions. Cannot remember why I have removed it again. I don't think optional extensions should go in `require-dev` though.

**Comment by aschempp** [comment]
> I don't think optional extensions should go in require-dev though.

No experience with that either. But theoretically it's the same as with other dev requirements, you need that if you want code autocompletition and successfully run tests.

**Comment by leofeyer** [comment]
Yeah, but optional extensions are no dev requirement. If at all, they belong in the "suggest" section.

**Comment by aschempp** [comment]
Not true. Optional dependencies or suggestions are also in require-dev usually because you want them for development.

**Comment by leofeyer** [comment]
There is no point in discussing this.

**Comment by leofeyer** [comment]
Aside from the discussion about optional extensions, which mandatory extensions should we add? Once @ausi's image service has been merged, the GDlib extension will no longer be mandatory, will it? So the DOM extension is actually the only mandatory one?

**Comment by leofeyer** [comment]
Ok, so here are the extensions we are actually using:

```json
{
    "require": {
        "php": ">=5.5.0",
        "ext-dom": "*",
        "ext-pcre": "*"
    },
    "require-dev": {
        "ext-fileinfo": "*"
    }
}
```

Don't know if it makes much sense to add these?

**Comment by discordier** [comment]
It does!

**Comment by ausi** [comment]
> Once @ausi's image service has been merged, the GDlib extension will no longer be mandatory, will it?

If PHP extensions that are used by deprecated classes count as required, GD has to stay because `Contao\GdImage` still exists.

**Comment by leofeyer** [comment]
It depends on whether we deprecate the class or not.

**Comment by ausi** [comment]
`GdImage` should be definitely deprecated IMO

**Comment by discordier** [comment]
Even if we deprecate it, we are still using it.
We can only remove the dependency on gdlib in 5.0 therefore (which sadly means we will have to introduce it first as it was simply forgotten so far).

**Comment by leofeyer** [comment]
Implemented in 70bc66ae3e16ac37c14f81d45ce2f0333726b304.
